### PR TITLE
feat: sync fork in Fink

### PR DESCRIPTION
**Pull request by NiklasBuchfink**
Adds fork syncing capabilities to Fink

1. Merge without conflicts

image

2. Merge with conflicts and actions in GitHub required

image

Loom:
https://www.loom.com/share/562c9b6b4bf84fc08cc47596e1fab570

Related issue: https://github.com/opral/monorepo/issues/1614

### CODE REVIEW

**Comment by NiklasBuchfink** [comment]
@NilsJacobsen You're right, I needed a second GitHub account for this :D

Can you provide me the image for the GitHub Sync UI, you had in mind?
image

**Comment by NiklasBuchfink** [comment]
@janfjohannes I have the LixFloat action in place.

image

Now only #1996 is missing to complete this feature.

File: inlang/source-code/editor/src/pages/@host/@owner/@repository/components/Gitfloat.tsx [hunk 1]
```diff
@@ -291,7 +316,35 @@ export const Gitfloat = () => {
...
+							Get new project changes
+							<sl-button
+								prop:size="small"
+								onClick={async () => {
+									setIsMerging(true)
+									await repo()?.mergeUpstream()
+									refetchRepo()
+									setIsMerging(false)
+									setTimeout(() => {
+										mutateForkStatus({ ...forkStatus(), behind: 0, conflicts: false })
```
Comment (janfjohannes) [inline inlang/source-code/editor/src/pages/@host/@owner/@repository/components/Gitfloat.tsx:333]: i assume this is an optimisstic update?

**Review by janfjohannes — APPROVED** [review]
looks cool! let me know if i should also do manual testing

File: inlang/source-code/editor/src/pages/@host/@owner/@repository/components/Gitfloat.tsx (hunk 1 above)
Comment (NiklasBuchfink) [reply inlang/source-code/editor/src/pages/@host/@owner/@repository/components/Gitfloat.tsx:333]: Yes, I can add the refetch of forkStatus(), but the value only changes once it's finished. Therefore, after the animation is finished, I mutate the value to the default.

**Pull request merged**